### PR TITLE
next: Display blunder bar in the winrate graph

### PR DESCRIPTION
**Pull request by zsalch**
#351 

I'm trying to display blunder bar in the winrate graph.

* 1. Add new option 'show-winrate-blunder-bar' in the config.txt to turn on blunder bar, default is turn off.
* 2. Can specify the minimum width of the blunder bar by the option 'minimum-winrate-blunder-bar-width' in the config.txt, default is 3.
* 3. Considering that the winrate change should be mostly below 50%, an option is added to weighted display the height of the bar, that is, the change below 50% increases the display height, and the above 50% reduces the height. Currently only a simple weighted, the winrate change below 50% uses 75% of the display area, the above uses 25% of the display area. The default is off

turn off -> turn on
image
Adjusted for the green of the winrate graph should always be on top:
image

* 4. Due to the small width of the winrate graph, when the move number is large, the display is crowded, so an option 'dynamic-winrate-graph-width' is added to dynamically adjust the width, so that can see clearly when the number of steps is small. Default is turn off.

* 5. Add new option 'winrate-stroke-width' in the config.txt to set the winrate line width, default is 3.

* 6. Add large winrate mode, press Ctrl+W to toggle:
lb

### CODE REVIEW

**Comment by featurecat** [comment]
thanks. I think the green of the winrate graph should always be on top. is it possible to make it more visible? it's confusing in the middle of your graphs.

**Comment by featurecat** [comment]
also I absolutely love the name "blunder bar" 😁

**Comment by zsalch** [comment]
I'm confuse too.
Need a solution to set the drawline and fillrect z-order.

**Comment by zsalch** [comment]
Adjusted:
image

File: src/main/java/featurecat/lizzie/gui/LizzieFrame.java [hunk 1]
```diff
@@ -334,117 +352,286 @@ public void paint(Graphics g0) {
...
+          vh = panelH;
+          // subboard
+          subBoardY = topInset;
+          subBoardWidth = panelW - leftInset;
+          subBoardHeight = panelH;
+          subBoardLength = Math.min(subBoardWidth, subBoardHeight);
+          subBoardX = statx + (vw - subBoardLength) / 2;
+        }
+      } else {
+        // Portrait mode
```
Comment (OlivierBlanvillain) [inline src/main/java/featurecat/lizzie/gui/LizzieFrame.java]: It looks like some code from #398 leaked into this PR. Could you try to clean this up before the review?

Comment (zsalch) [reply src/main/java/featurecat/lizzie/gui/LizzieFrame.java]: It not as a portrait mode, but just a display improvement, which reduces the user's error when resizing the window.

**Comment by featurecat** [comment]
Should I include this in Lizzie 0.6, and should I wait until you're done reviewing to do so?

**Pull request merged**